src/containers/trends/TrendsContainer.js

```diff
@@ -154,48 +174,55 @@ export class TrendsContainer extends React.Component {
   }
 
   getCurrentDay() {
+    const { timePrefs } = this.props;
     const { dateDomain: { end } } = this.state;
-    // TODO: replace with more robust code for finding noon of the local timezone day
-    return new Date(Date.parse(end) - (864e5 / 2)).toISOString();
+    return datetime.localNoonBeforeTimestamp(
+      end,
+      datetime.getTimezoneFromTimePrefs(timePrefs)
+    ).toISOString();
   }
 
   setExtent(newDomain) {
     const { cbgByDate, smbgByDate } = this.props;
+    const { mostRecent } = this.state;
     this.refilterByDate(cbgByDate, newDomain);
     this.refilterByDate(smbgByDate, newDomain);
     this.setState({
       currentCbgData: cbgByDate.top(Infinity).reverse(),
       currentSmbgData: smbgByDate.top(Infinity).reverse(),
       dateDomain: { start: newDomain[0], end: newDomain[1] },
     });
+    this.props.onDatetimeLocationChange(newDomain, newDomain[1] >= mostRecent);
   }
 
   goBack() {
     const { dateDomain: { start: newEnd } } = this.state;
-    const start = utcDay.offset(new Date(newEnd), -this.props.extentSize).toISOString();
+    const { timePrefs } = this.props;
+    const timezone = datetime.getTimezoneFromTimePrefs(timePrefs);
+    const start = datetime.timezoneAwareOffset(newEnd, timezone, {
+      amount: -this.props.extentSize,
```

Comment:
maybe this just reads strangely but I did a double take so just thought I would point it out